core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -394,21 +405,21 @@ class GroupCoordinator(val brokerId: Int,
       partitions.map {case topicAndPartition => (topicAndPartition, OffsetMetadataAndError.GroupCoordinatorNotAvailable)}.toMap
     } else if (!isCoordinatorForGroup(groupId)) {
       partitions.map {case topicAndPartition => (topicAndPartition, OffsetMetadataAndError.NotCoordinatorForGroup)}.toMap
+    } else if (isCoordinatorLoadingInProgress(groupId)) {
+      partitions.map {case topicAndPartition => (topicAndPartition, OffsetMetadataAndError.GroupLoading)}.toMap
     } else {
       // return offsets blindly regardless the current group state since the group may be using
       // Kafka commit storage without automatic group management
-      offsetManager.getOffsets(groupId, partitions)
+      groupManager.getOffsets(groupId, partitions)
     }
   }
 
   def handleGroupImmigration(offsetTopicPartitionId: Int) = {
-    // TODO we may need to add more logic in KAFKA-2017
-    offsetManager.loadOffsetsFromLog(offsetTopicPartitionId)
+    groupManager.loadGroupsForPartition(offsetTopicPartitionId)
```

Comment:
Ack. We will let the GroupMetadataManager itself maintain the owned partitions, which are going to be synced with the same loading partition object.